src/renderer/components/+namespaces/namespace-select-filter.tsx

```diff
@@ -49,14 +61,22 @@ const Placeholder = observer((props: PlaceholderProps<any, boolean>) => {
   };
 
   return (
-    <components.Placeholder {...props}>
+    <components.Placeholder {...props} data-testid="namespace-select-filter-placeholder">
       {getPlaceholder()}
     </components.Placeholder>
   );
 });
 
+export interface NamespaceSelectFilterProps extends NamespaceSelectProps {
+  maxItems?: number;
+}
+
+interface NamespaceSelectFilterDependencies {
+  namespaceStore: NamespaceStore,
+}
+
```

Comment:
Maybe better to put `namespaceStore` inside `NamespaceSelectFilterProps`?
Or at least extend `NamespaceSelectFilterProps` with `NamespaceSelectFilterDependencies` because it's anyway  accessed via component's `props`, right?

Moreover you put it already in props interface above for placeholder:

```ts
interface NamespaceSelectFilterPlaceholderProps {
  namespaceStore: NamespaceStore
}
```

Wdyt?

Reply:
No, this is the correct usage so that dependencies are not exported to the extension API or the documentation.